src/diffusers/pipelines/text_to_video_synthesis/pipeline_text_to_video_zero.py

```diff
@@ -0,0 +1,521 @@
+
+                # compute the previous noisy sample x_t -> x_t-1
+                latents = self.scheduler.step(noise_pred, t, latents, **extra_step_kwargs).prev_sample
+                # latents = latents - alpha * grads / (torch.norm(grads) + 1e-10)
+                # call the callback, if provided
+
+                if i < len(timesteps) - 1 and timesteps[i + 1] == t0:
+                    x_t0_1 = latents.detach().clone()
+                    print(f"latent t0 found at i = {i}, t = {t}")
+                elif i < len(timesteps) - 1 and timesteps[i + 1] == t1:
+                    x_t1_1 = latents.detach().clone()
+                    print(f"latent t1 found at i={i}, t = {t}")
+
+                if i == len(timesteps) - 1 or ((i + 1) > num_warmup_steps and (i + 1) % self.scheduler.order == 0):
+                    progress_bar.update()
+                    if callback is not None and i % callback_steps == 0:
+                        callback(i, t, latents)
+
+        latents = rearrange(latents, "(b f) c w h -> b c f  w h", f=f)
+
+        res = {"x0": latents.detach().clone()}
+        if x_t0_1 is not None:
+            x_t0_1 = rearrange(x_t0_1, "(b f) c w h -> b c f  w h", f=f)
+            res["x_t0_1"] = x_t0_1.detach().clone()
+        if x_t1_1 is not None:
+            x_t1_1 = rearrange(x_t1_1, "(b f) c w h -> b c f  w h", f=f)
+            res["x_t1_1"] = x_t1_1.detach().clone()
+        return res
+
+    def decode_latents(self, latents):
+        latents = 1 / 0.18215 * latents
+        latents = rearrange(latents, "b c f h w -> (b f) c h w")
+        video = self.vae.decode(latents).sample
+        video = rearrange(video, "b c h w -> b h w c")
+        video = (video / 2 + 0.5).clamp(0, 1)
+        # we always cast to float32 as this does not cause significant overhead and is compatible with bfloa16
+        video = video.cpu().float().numpy()
+        return video
+
+    @torch.no_grad()
+    def __call__(
+        self,
+        prompt: Union[str, List[str]],
+        video_length: Optional[int],
+        height: Optional[int] = None,
+        width: Optional[int] = None,
+        num_inference_steps: int = 50,
+        guidance_scale: float = 7.5,
+        negative_prompt: Optional[Union[str, List[str]]] = None,
+        num_videos_per_prompt: Optional[int] = 1,
+        eta: float = 0.0,
+        generator: Optional[Union[torch.Generator, List[torch.Generator]]] = None,
+        latents: Optional[torch.FloatTensor] = None,
+        motion_field_strength_x: float = 12,
+        motion_field_strength_y: float = 12,
+        output_type: Optional[str] = "tensor",
+        return_dict: bool = True,
+        callback: Optional[Callable[[int, int, torch.FloatTensor], None]] = None,
+        callback_steps: Optional[int] = 1,
+        use_motion_field: bool = True,
+        inject_noise_to_warp: bool = False,
+        t0: int = 45,
+        t1: int = 48,
+        **kwargs,
```

Comment:
Do we need `**kwargs` here? Would be maybe just add `frame_ids` ?

Reply:
Done!